pbs/its
# OCTO-9149: don't turn transparent pngs into jpegs

locking our pillow version to 5.3.0 to avoid https://github.com/python-pillow/Pillow/issues/3557 - it sounds like pillow has a planned release for April that will solve this problem.

As of this patch, using the `format=auto` query parameter will only convert a png to a jpeg if the png doesn't support an alpha channel (transparency: http://www.libpng.org/pub/png/book/chapter08.html).

# To test locally

* check this branch out: `git checkout feature/OCTO-9149-auto-should-preserve-transparent-pngs`
* start the application: `docker-compose up --build`
* open http://localhost:5000/tests/images/transparent_complex.png?format=auto

the image should remain a png with a transparent background: this is correct behavior.

* check out former application code (to see previous behavior): `git checkout master -- its/optimize.py`
* open http://localhost:5000/tests/images/transparent_complex.png?format=auto

you should see the previous behavior: the image becomes a jpeg and loses its transparency.
